Makefile

```diff
@@ -58,29 +60,46 @@ release:
 	./hack/release.sh
 .PHONY: release
 
-provisioner-%: 
-	$(MAKE) ARCH=$* provisioner
-
-provisioner:
-	mkdir -p _output
-	# because COPY does not expand build arguments, we need substitute it
-	cat ./deployment/docker/Dockerfile \
-		| sed "s|QEMUARCH|$(QEMUARCH)|g" \
-		> $(TEMP_DIR)/Dockerfile
-ifneq ($(ARCH),amd64)
-	# Register /usr/bin/qemu-ARCH-static as the handler for non-x86 binaries in the kernel
-	$(SUDO) ./third_party/multiarch/qemu-user-static/register/register.sh --reset
-endif
-	docker build -t $(MUTABLE_IMAGE) --build-arg GOVERSION=$(GOVERSION) --build-arg ARCH=$(ARCH) -f $(TEMP_DIR)/Dockerfile .
-	docker tag $(MUTABLE_IMAGE) $(IMAGE)
-	rm -rf $(TEMP_DIR)
-.PHONY: provisioner
-
-test: provisioner
+# used in `make test` and `make e2e`
+# builds without pushing to the registry
+build-container-linux-%:
+	$(DOCKER) buildx build --file=./deployment/docker/Dockerfile --platform=linux/$* \
+		-t $(STAGINGIMAGE):$(STAGINGVERSION)_linux_$* \
+		--build-arg OS=linux \
+		--build-arg ARCH=$* \
+		--build-arg BUILDPLATFORM=linux \
+		--build-arg STAGINGVERSION=$(STAGINGVERSION) .
+
+build-and-push-container-linux-%: init-buildx
+	$(DOCKER) buildx build --file=./deployment/docker/Dockerfile --platform=linux/$* \
+		-t $(STAGINGIMAGE):$(STAGINGVERSION)_linux_$* \
+		--build-arg OS=linux \
+		--build-arg ARCH=$* \
+		--build-arg BUILDPLATFORM=linux \
+		--build-arg STAGINGVERSION=$(STAGINGVERSION) --push .
+
+build-and-push-container-windows-%: init-buildx
+	$(DOCKER) buildx build --file=./deployment/docker/Dockerfile.Windows --platform=windows/amd64 \
+		-t $(STAGINGIMAGE):$(STAGINGVERSION)_windows_$* \
+		--build-arg BASE_IMAGE=$(call lookup,$*,$(WINDOWS_DISTROS),$(WINDOWS_BASE_IMAGES)) \
+		--build-arg STAGINGVERSION=$(STAGINGVERSION) --push .
+
+test: build-container-linux-amd64
```

Comment:
the test is just for amd?

Reply:
yes, the unit test only runs in linux amd64